### PR TITLE
Don't re-index when upstream index is unchanged

### LINKED ISSUE #113: Registry indexing appears to be broken
Registry indexing appears to be broken, since several recent changes aren't being picked up:
https://github.com/buildpacks/registry-index/pull/5767#issuecomment-1599443267
https://github.com/buildpacks/registry-index/issues/5773#issuecomment-1599309810

See:
https://cloud-native.slack.com/archives/C032YE21V1T/p1687338079300959

### PR DESCRIPTION
**Pull request by joshwlewis**
This PR aims to greatly reduce the number of unnecessary docker pulls and postgres writes we make. We don't need to rebuild the entire `buildpacks` table and re-fetch every docker image if there have been no new changes to the upstream index. In this PR, we keep track of the last commit sha of that we fully indexed, and skip the process entirely if the sha hasn't changed.

This should lower the number of times we perform a full reindex each day from greater than 100 to less than 10. This should lower our impact on upstream registries (like dockerhub and ECR) and make rate limiting less frequent.

I've also changed the delay between re-indexing from 5 minutes to 1.5 minutes, since most of the time, there won't be a new commit, so there won't be anything to do. This should reduce the time for a change to appear in the registry api / website by a few minutes.

Related #114 
Fixes #113

### CODE REVIEW

File: bin/indexer [hunk 1]
```diff
@@ -4,24 +4,30 @@ const fs = require('fs');
 const execa = require('execa');
 const path = require('path');
 const os = require('os');
+const reindex_delay = process.env.REINDEX_DELAY || 300000;
```
Comment (edmorley) [inline bin/indexer:7]: I suppose we could even set this to something lower than 5 mins now (eg 1-2 mins?) to reduce release latency (xref https://github.com/heroku/languages-github-actions/pull/28#discussion_r1238940026)

**Review by edmorley — APPROVED** [review]
Thank you!

**Review by hone — APPROVED** [review]

File: bin/indexer (hunk 1 above)
Comment (joshwlewis) [reply bin/indexer:7]: Yeah, I'll go adjust this.

Comment (joshwlewis) [reply bin/indexer]: https://github.com/buildpacks/registry-api/pull/115/commits/76b95c24c0995ef051b445b1dbcf35b96423e882

**Pull request merged**